cmd/kubeadm/app/phases/addons/proxy/proxy.go

```diff
@@ -56,11 +60,52 @@ func EnsureProxyAddon(cfg *kubeadmapi.MasterConfiguration, client clientset.Inte
 		return err
 	}
 
-	proxyConfigMapBytes, err := kubeadmutil.ParseTemplate(KubeProxyConfigMap, struct{ MasterEndpoint string }{
-		// Fetch this value from the kubeconfig file
-		MasterEndpoint: masterEndpoint})
+	// Validate the kube-proxy bind address
+	bindAddress := cfg.KubeProxy.Config.BindAddress
+	if net.ParseIP(bindAddress) == nil {
+		return fmt.Errorf("kube-proxy bind address '%s' is not a valid IP address", bindAddress)
+	}
+
+	// If the bind address begins with a ':' (e.g. the IPv6 address '::'),
+	// then wrap it in double quotes; otherwise the config parser will
+	// consider it a syntax error.
+	if bindAddress[0] == ':' {
+		bindAddress = "\"" + bindAddress + "\""
+	}
+
+	serial, err := kubeadmutil.MarshalToYamlForCodecsWithShift(cfg.KubeProxy.Config, kubeproxyconfig.SchemeGroupVersion,
+		kubeproxyconfigscheme.Codecs)
 	if err != nil {
-		return fmt.Errorf("error when parsing kube-proxy configmap template: %v", err)
+		return fmt.Errorf("error when marshaling: %v", err)
+	}
+	// Parse the given kubernetes version
+	k8sVersion, err := version.ParseSemantic(cfg.KubernetesVersion)
+	if err != nil {
+		return fmt.Errorf("couldn't parse kubernetes version %q: %v", cfg.KubernetesVersion, err)
+	}
+	var proxyConfigMapBytes []byte
+	if k8sVersion.LessThan(version.MustParseSemantic("v1.9.0")) {
```

Comment:
add a new version constant in `cmd/kubeadm/app/constants.MinimumKubeProxyComponentConfigVersion` as `v1.9.0-alpha.3`

Reply:
also switch the logic here to be `k8sVersion.AtLeast(kubeadmconstants.MinimumKubeProxyComponentConfigVersion)` and make the v1.9 case the first one